### PR TITLE
Guard Symbol and ShapeGuardPrinter behind HAS_SYMPY

### LINKED ISSUE #90696: Pytorch nightly conda builds for Linux and Windows failing
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/actions/runs/3673791849/jobs/6213742922
Failure:
```
+ python -c 'import torch'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/__init__.py", line 1232, in <module>
    from . import _meta_registrations
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/_meta_registrations.py", line 7, in <module>
    from torch._decomp import _add_op_to_registry, global_decomposition_table, meta_table
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/_decomp/__init__.py", line 168, in <module>
    import torch._decomp.decompositions
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/_decomp/decompositions.py", line 10, in <module>
    import torch._prims as prims
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/_prims/__init__.py", line 34, in <module>
    from torch.fx.experimental.symbolic_shapes import sym_float
  File "/opt/conda/envs/testenv/lib/python3.9/site-packages/torch/fx/experimental/symbolic_shapes.py", line 451, in <module>
    class Symbol(sympy.Dummy):
NameError: name 'sympy' is not defined
```

### Versions

nightly

cc @ezyang @gchanan @zou3519 @seemethere @malfet

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eliuriegas@meta.com>

Follow up to https://github.com/pytorch/pytorch/pull/90528

Fixes https://github.com/pytorch/pytorch/issues/90696

### CODE REVIEW

**Review by weiwangmeta — APPROVED** [review]
LGTM

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -446,34 +446,35 @@ def wrapper(self, *args, **kwargs):
...
-        symbol_to_source,
-    ):
-        super().__init__()
-        self.symbol_to_source = symbol_to_source
-
-    def _print_Symbol(self, expr) -> str:
-        assert isinstance(expr, Symbol), str(type(expr))
-        assert expr in self.symbol_to_source, f"{expr} (could be from {expr.snames}) not in {self.symbol_to_source}"
-        return self.symbol_to_source[expr][0]
+if HAS_SYMPY:
```
Comment (atalman) [inline torch/fx/experimental/symbolic_shapes.py:449]: is this system env variable ?

**Review by atalman — APPROVED** [review]
LGTM

File: torch/fx/experimental/symbolic_shapes.py (hunk 1 above)
Comment (seemethere) [reply torch/fx/experimental/symbolic_shapes.py:449]: It's declared at the top of the file near the sympy import

Comment (seemethere) [reply torch/fx/experimental/symbolic_shapes.py:449]: https://github.com/pytorch/pytorch/blob/a6cd8e23ac2514ccb60ded555568622792fcbd0d/torch/fx/experimental/symbolic_shapes.py#L19-L25

**Review by malfet — APPROVED** [review]
LGTM, but considering that SYMPY is now a mandatory dep, should we really have this guard?

**Pull request closed without merging**